ui/src/StateContainer.js

```diff
@@ -172,15 +173,28 @@ export class StateContainer extends LitElement {
         return html` <vox-toast message="${this.toast}"></vox-toast> `;
     }
 
+    updateCollectionState(e) {
```

Comment:
You can handle the state transition in the `updated` function

```javascript
updated(changedProperties) {
  if (changedProperties.has('collectionState') && this.collectionState === CollectionStates.TRANSITIONING) {
    this.handleChangePrompt();
  }
}

updateCollectionState(e) {
  this.collectionState = e.detail.state;
}

handleSkipPrompt() {
  this.collectionState = CollectionStates.TRANSITIONING;
}
```

https://lit-element.polymer-project.org/guide/lifecycle#updated